### PR TITLE
Add httpd conf parsers/combiners for httpd24 and jbcs-httpd24 SCLs

### PR DESCRIPTION
**Pull request by matysek**
httpd config files are for Apache HTTP server in different locations when it is installed from Software Collections.

### CODE REVIEW

File: insights/combiners/httpd_conf.py [hunk 1]
```diff
@@ -66,6 +66,7 @@
 
 from insights.core.plugins import combiner
 from insights.parsers.httpd_conf import HttpdConf, dict_deep_merge, ParsedData
+from insights.parsers.httpd_conf import HttpdConfSclHttpd24, HttpdConfSclJbcsHttpd24
 
 
 @combiner(HttpdConf)
```
Comment (jsvob) [inline insights/combiners/httpd_conf.py]: I *think* that the common class from which all three resulting combiners inherit should *not* be decorated with HttpdConf.

Reasons:

1. Each combiner should process exactly one set of config files. The combiner doesn't work correctly with more than 1 httpd.conf (the main file that contains the include most probably).

2. I remember this being a problem in the past. IDK whether it is problematic now.

File: insights/parsers/httpd_conf.py [hunk 2]
```diff
@@ -186,6 +186,30 @@ def add_to_dict_list(dictionary, key, element):
                         add_to_dict_list(where_to_store, option, parsed_data)
 
 
+@parser(Specs.httpd_conf_scl_httpd24)
+class HttpdConfSclHttpd24(HttpdConf):
```
Comment (jsvob) [inline insights/parsers/httpd_conf.py]: The same note about inheritance applies here.

**Comment by bfahr** [comment]
@csams would you mind to take a look at this and make sure it is consistent with your new configparser code.

**Comment by bfahr** [comment]
@matysek could you please update your PR to fix the merge conflict.

**Comment by matysek** [comment]
@bfahr @csams becase of recent changes to httpd_conf parser/combiner it causes hard merge conflicts that cannot be easily resolved and I am not sure what is the preferred way to integrate these SCL parsers. Could you take a look at it?

**Comment by jsvob** [comment]
@matysek You asked me to look at this. I did a diff of the parsers/httpd_conf.py and combiners/httpd_conf.py between this PR and current master. It seems to me that the diffing tools mistakenly think there is a collision. I think that if you redo the changes you did on a fresh new branch branched off master, you will find that actually, it's pretty independent and you can do pretty much the exact same thing. I also think that wading through the <<<< / >>>> noise of a git merge/git rebase would be chaotic, maybe this is where you stopped? 

But then again, I didn't actually try to rebase this branch and you did, so maybe I don't see something crucial here.

**Comment by bfahr** [comment]
I spoke with @matysek yesterday and he is working on this.

**Comment by matysek** [comment]
@csams Any progress on reviewing this PR? There are two more tasks that depends on this PR. Thanks.

File: insights/combiners/httpd_conf.py [hunk 3]
```diff
@@ -382,6 +382,56 @@ def conf_path(self):
...
+class HttpdConfSclHttpd24Tree(ConfigCombiner):
+    """
+    Exposes httpd configuration Software Collection httpd24 through the configtree
+    interface. Correctly handles all include directives.
+
+    See the :py:class:`insights.core.ConfigComponent` class for example usage.
+    """
+    def __init__(self, confs):
+        includes = startswith("Include")
+        super(HttpdConfSclHttpd24Tree, self).__init__(confs, "httpd.conf", includes)
```
[the remark this answers is not in the document]
Comment (csams) [reply insights/combiners/httpd_conf.py]: @bfahr "http.conf" here means to use the parser for the file with that basename as the root of the config.

File: insights/combiners/httpd_conf.py [hunk 4]
```diff
@@ -382,6 +382,56 @@ def conf_path(self):
...
+class HttpdConfSclJbcsHttpd24Tree(ConfigCombiner):
+    """
+    Exposes httpd configuration Software Collection jbcs-httpd24 through the configtree
+    interface. Correctly handles all include directives.
+
+    See the :py:class:`insights.core.ConfigComponent` class for example usage.
+    """
+    def __init__(self, confs):
+        includes = startswith("Include")
+        super(HttpdConfSclJbcsHttpd24Tree, self).__init__(confs, "httpd.conf", includes)
```
[the remark this answers is not in the document]
Comment (csams) [reply insights/combiners/httpd_conf.py]: @bfahr "http.conf" here means to use the parser for the file with that basename as the root of the config.

**Comment by matysek** [comment]
@csams I checked and there are cases where they are installed both - httpd in default location and httpd24-httpd from sw collections. So it is not applicable to just extend existing datasources.

**Pull request merged**